programs/fileio.c

```diff
@@ -1003,6 +1006,10 @@ FIO_compressZstdFrame(FIO_prefs_t* const prefs,
     /* init */
     if (fileSize != UTIL_FILESIZE_UNKNOWN) {
         CHECK(ZSTD_CCtx_setPledgedSrcSize(ress.cctx, fileSize));
+    } else if (prefs->streamSrcSize > 0) {
+      /* unknown source size; use the declared stream size and disable writing this size to frame during compression */
+      CHECK( ZSTD_CCtx_setPledgedSrcSize(ress.cctx, prefs->streamSrcSize) );
+      CHECK( ZSTD_CCtx_setParameter(ress.cctx, ZSTD_c_contentSizeFlag, 0) );
```

Comment:
We don't need to disable the content size flag here, since the stream size must be exact otherwise compression will error. So we can pick which behavior we prefer. I think we should leave `contentSizeFlag` enabled (the default), since this is the only way to tell zstd to write the content size when the source file size is unknown. This is consistent with the zstd CLI always writing in the pledged source size when it is known.

Reply:
That's true, didn't think about that